### PR TITLE
SEAB-5021:  Confirm support for CWL Version 1.2

### PR DESCRIPTION
**Pull request by svonworl**
**Description**
This PR adds a couple of tests that help to confirm CLI compatibility with CWL version 1.2.

Per the spec, CWL changed as follows between version 1.1 and 1.2:
https://www.commonwl.org/v1.2/Workflow.html#Changelog
Not much is different.

Knowing how we parse and a bit about how the CLI is structured, my theory was that if our CLI supports CWL v1.1, and the CLI could process a v1.2 workflow that contained either a "when" clause or an "Operation", that it probably can successfully process a very large proportion of v1.2 workflows.  So, I added a test corresponding to each of those two cases.

As it turns out, the CLI processes a workflow with a step containing an "Operation" with no problems, but when cwltool is invoked to run it, it fails because an "Operation" is defined as a placeholder with undefined semantics which cannot be executed:
https://www.commonwl.org/v1.2/Workflow.html#Operation
In other words, this is expected behavior.

**Review Instructions**
Confirm the presence of the new tests, and that they pass.

**Issue**
https://ucsc-cgl.atlassian.net/browse/SEAB-5021

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that you pass the basic style checks and unit tests by running `./mvnw clean install` 
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.

### CODE REVIEW

**Review by coverbeck — APPROVED** [review]

**Comment by svonworl** [comment]
> > An implementation (or other tooling) may provide a means of binding an Operation to a concrete process (such as Workflow, CommandLineTool, or ExpressionTool) with a compatible signature.
> 
> Interesting find, did you find any examples? Is this like CWL having its own concept of inheritance? If so, would it be worthwhile testing a CWL workflow with a concrete process that implements(?) an operation?

I _think_ an Operation is more like a placeholder:
https://www.commonwl.org/user_guide/topics/operations.html

So, I'd say no, we probably don't need another test because a fully-cooked, reproducible workflow shouldn't contain Operations.  If/when we tighten up CWL validation in the webservice, we might want to consider marking a workflow as invalid if it contains an Operation step, perhaps.

**Review by denis-yuen — APPROVED** [review]

**Review by kathy-t — APPROVED** [review]
